src/diffusers/loaders/peft.py

```diff
@@ -230,8 +238,12 @@ def load_lora_adapter(
             raise ValueError("`network_alphas` cannot be None when `prefix` is None.")
 
         if prefix is not None:
+            metadata = state_dict.pop(LORA_ADAPTER_METADATA_KEY, None)
             state_dict = {k[len(f"{prefix}.") :]: v for k, v in state_dict.items() if k.startswith(f"{prefix}.")}
 
+            if metadata is not None:
+                state_dict[LORA_ADAPTER_METADATA_KEY] = metadata
+
```

Comment:
Would it not be easier to add `and if k != LORA_ADAPTER_METADATA_KEY` to the dict comprehension instead of popping and re-adding the meta data?

Reply:
Rewrote this a bit as the metadata related bits can be moved out of `if prefix is not None`.